Features/NotesFeature/NotesViewController.swift

```diff
@@ -47,7 +47,22 @@ final class NotesViewController: UIHostingController<NotesView> {
             editMode: Binding(
                 get: { [weak self] in self?.currentEditMode },
                 set: { [weak self] value in self?.viewModel.editMode = value ?? .inactive }
-            )
-        )
+            ),
+            customItems: [
+                UIBarButtonItem(
+                    image: UIImage(systemName: "square.and.arrow.up"),
+                    style: .plain,
+                    target: self,
+                    action: #selector(shareAllNotes))])
+    }    
+
+    @objc private func shareAllNotes() {
+        Task {
+            let notesText = await viewModel.prepareNotesForSharing()
+            let activityViewController = UIActivityViewController(activityItems: [notesText], applicationActivities: nil)
+            activityViewController.popoverPresentationController?.sourceView = self.view
```

Comment:
I think this might crash on iPads or the popover arrow might not be pointing to the share button. Could you please verify that?

Reply:
You were right, added some code from [here](https://github.com/quran/quran-ios/blob/4df3eb4a8c91571c18d9e8e38592bf7a0bad6b50/Features/SettingsFeature/UIViewController%2BShare.swift#L19-L24) to fix it.
![simulator_screenshot_E0926277-A1E8-42F2-865F-0533B949504F](https://github.com/user-attachments/assets/72d42f89-f8c1-4b80-ade0-e015c5ed4a61)